pennylane/ftqc/graph_state_preparation.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""This module contains the GraphStatePrep template."""
+
+import pennylane as qml
+from pennylane.operation import Operation
+
+from .qubit_graph import QubitGraph
+
+
+class GraphStatePrep(Operation):
+    r"""
+    Encode a graph state with the specified lattice structure, operations on each qubit, entanglement operations for nearest qubits and qubit graph.
+    The initial graph is :math:`|0\rangle^{V}`, given each qubit in the graph is in the :math:`|0\rangle` state and is not entangled with each other.
+    The target graph state :math:`| \psi \rangle = \prod\limits_{\{a, b\} \in E} U_{ab}|+\rangle^{V}` is prepared as below:
+    1. Each qubit is prepared as :math:`|+\rangle^{V}` state by applying the `qubit_ops` (`Hadamard` gate) operation.
+    2. Entangle every nearest qubit pair in the graph with `entanglement_ops` (`CZ` gate) operation.
+
+    Args:
+        qubit_graph (QubitGraph): QubitGraph object mapping qubit to wires.
+        qubit_ops (Operations): Operator to prepare the initial state of each qubit. Default as ``qml.H``.
+        entanglement_ops (Operations): Operator to entangle nearest qubits. Default as ``qml.CZ``.
```

Comment:
I think the conclusion after Joey's comment on my last PR was we don't actually need the type in parentheses after, since there are type hints in the call signature, so maybe they can actually just be deleted.

```suggestion
        qubit_ops (Operation): Operator to prepare the initial state of each qubit. Default as ``qml.H``.
        entanglement_ops (Operation): Operator to entangle nearest qubits. Default as ``qml.CZ``.
```